File: android/src/main/kotlin/com/screenmeet/sdk_live_flutter_plugin/handlers/FeatureRequestStreamHandler.kt
package com.screenmeet.sdk_live_flutter_plugin.handlers

import com.screenmeet.sdk.Entitlement
import com.screenmeet.sdk.Feature
import com.screenmeet.sdk_live_flutter_plugin.zipFeature
import com.screenmeet.sdk_live_flutter_plugin.zipFeatureCancel
import io.flutter.plugin.common.EventChannel

class FeatureRequestStreamHandler: EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) { sink = events }
    override fun onCancel(arguments: Any?) { sink = null }

    fun sendPermissionRequest(feature: Feature) {
        sink?.success(zipFeature(feature))
    }

    fun sendFeatureCancel(entitlement: Entitlement) {
        sink?.success(zipFeatureCancel(entitlement))
    }

}

File: ios/Classes/StreamHandlers/FeatureRequestStreamHandler.swift
//
//  PermissionRequestStreamHandler.swift
//  sdk_live_flutter_plugin
//
//  Created by Ross on 04.02.2022.
//

import Foundation
import Flutter
import ScreenMeetSDK

class FeatureRequestStreamHandler: NSObject, FlutterStreamHandler {
    private let pm = SwiftChannelParamsManager()

    var sink: FlutterEventSink?
       
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
       
    func sendPersmissionRequest(_ feature: SMFeature) {
        guard let sink = sink else { return }
        sink(pm.featureToDict(feature))
    }
       
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}
